pkg/auth/auth_test.go

```diff
@@ -104,10 +104,39 @@ func TestParseCredentialsKey(t *testing.T) {
 			expectedRegistry:   "docker.io",
 		},
 		{
-			name:               "with http[s] prefix",
+			name:               "with http[s] prefix - accept repositories",
 			arg:                "https://quay.io",
 			acceptRepositories: true,
-			expectedKey:        "",
+			expectedKey:        "quay.io",
+			expectedRegistry:   "quay.io",
+		},
+		{
+			name:               "with http[s] prefix",
+			arg:                "http://example.com/v2/",
+			acceptRepositories: true,
+			expectedKey:        "example.com",
+			expectedRegistry:   "example.com",
+		},
+		{
+			name:               "with http[s] prefix + path - accept repositories",
+			arg:                "https://quay.io/repo/imag:tag",
+			acceptRepositories: true,
+			expectedKey:        "quay.io",
+			expectedRegistry:   "quay.io",
+		},
+		{
+			name:               "with http[s] prefix + port",
+			arg:                "https://quay.io:1234",
+			acceptRepositories: true,
+			expectedKey:        "quay.io:1234",
+			expectedRegistry:   "quay.io:1234",
+		},
+		{
+			name:               "with http[s] prefix + port + path",
+			arg:                "https://quay.io:1234/repo@foo",
+			acceptRepositories: true,
+			expectedKey:        "quay.io:1234",
+			expectedRegistry:   "quay.io:1234",
 		},
 		{
 			name:               "failure with tag",
```

Comment:
Non-blocking: More tests with `acceptRepositories: false` would also be nice (at least https://host, https://host:port)